app/assets/javascripts/zpotlight/blocks/mirador_block.js

```diff
@@ -22,13 +34,82 @@ SirTrevor.Blocks.Mirador = (function() {
       MiradorWidgetAdmin.init();
     },
 
+    /**
+     * Overridden from https://github.com/projectblacklight/spotlight/blob/master/app/assets/javascripts/spotlight/blocks/resources_block.js
+     * to add the updateHiddenMiradorConfig
+     */
+    createItemPanel: function(data) {
+      var panel = this._itemPanel(data);
+      $(panel).appendTo($('.panels > ol', this.inner));
+      MiradorWidgetBlock.updateHiddenMiradorConfig($(this.el));
+      $('[data-behavior="nestable"]', this.inner).trigger('change');
+    },
+
+    /**
+     * Overridden from https://github.com/projectblacklight/spotlight/blob/master/app/assets/javascripts/spotlight/blocks/resources_block.js
+     * to trigger the 'item-removed'
+     */
+    afterPanelDelete: function() {
+      var block = $(this).closest('[data-behavior="mirador-widget"]');
+      var panel = $(this).closest('.field');
+      block.trigger('item-removed', { block: block, panel: panel });
+    },
+
+    /**
+     * A simplification from https://github.com/projectblacklight/spotlight/blob/master/app/assets/javascripts/spotlight/blocks/resources_block.js
+     */
+    _itemPanel: function(data) {
```

Comment:
This seems nearly identical to the code over in `MiradorWidgetAdmin` https://github.com/sul-dlss/vatican_exhibits/blob/cb212b58e7b2939fc43c13888dd4957d4ded8cff/app/assets/javascripts/mirador_widget_admin.js#L148-L167

Should these be re-using each-other in someway?